File: aoc_input.py
import requests
from dotenv import load_dotenv
import os

def get_aoc_input(day, year=2024):
    load_dotenv()

    session_cookie = os.getenv("AOC_SESSION_COOKIE")
    
    if session_cookie is None:
        raise ValueError("AOC_SESSION_COOKIE environment variable not set")
    
    cookies = {
        "session": session_cookie
    }

    url = f"https://adventofcode.com/{year}/day/{day}/input"

    response = requests.get(url, cookies=cookies)

    if response.status_code == 200:
        puzzle_input = response.text.strip()
        return puzzle_input
    else:
        raise ValueError(f"Failed to get input. Status code: {response.status_code}")

File: aoc_input/aoc_input.go
package aoc_input

import (
	"errors"
	"fmt"
	"io/ioutil"
	"net/http"
	"os"

	"github.com/joho/godotenv"
)

func GetAocInput(day int, year int) (string, error) {
	err := godotenv.Load("../.env")
	if err != nil {
		return "", fmt.Errorf("failed to load .env file: %v", err)
	}

	sessionCookie := os.Getenv("AOC_SESSION_COOKIE")
	if sessionCookie == "" {
		return "", errors.New("AOC_SESSION_COOKIE environment variable not set")
	}
	url := fmt.Sprintf("https://adventofcode.com/%d/day/%d/input", year, day)

	client := &http.Client{}
	req, err := http.NewRequest("GET", url, nil)
	if err != nil {
		return "", err
	}
	req.AddCookie(&http.Cookie{Name: "session", Value: sessionCookie})

	resp, err := client.Do(req)
	if err != nil {
		return "", err
	}
	defer resp.Body.Close()

	if resp.StatusCode != http.StatusOK {
		return "", fmt.Errorf("failed to fetch input: status code %d", resp.StatusCode)
	}

	body, err := ioutil.ReadAll(resp.Body)
	if err != nil {
		return "", err
	}

	return string(body), nil
}